src/System-Support/UndeclaredHolder.class.st

```diff
@@ -0,0 +1,12 @@
+"
+I am a dictionary holding UndeclaredVariables in the system. UndeclaredVariables represent the usage of a class that is not (yet) present in the system.
+
+I am named UndeclaredHolder instead of UndeclaredDictionary because one goal is to transform me into a weak collection in the future so that we do not have to explicitly clean up the undeclareds and delegate the work to the GC.
+"
+Class {
+	#name : 'UndeclaredHolder',
+	#superclass : 'IdentityDictionary',
```

Comment:
It would be better to not inherit dictionary but to wrap it with a dedicated API to register and lookup vars (instead of general collection methods). But it can be achieved later I guess.
Also what about different name? Something like UndeclaredRegistry.

Reply:
Indeed, just a tiny step of many.. this could even end up being a class variable on UndeclatedVariable (one global less!). But step by step.